src/api/app/services/file_service/uploader.rb

```diff
@@ -0,0 +1,114 @@
+module FileService
+  class Uploader
+    attr_reader :errors, :added_files
+
+    def initialize(package, files, empty_files, file_urls, comment)
+      @commit_filelist = []
+      @errors = []
+      @added_files = []
+
+      @package = package
+      @files = files
+      @empty_files = empty_files
+      @file_urls = file_urls
+      @comment = comment
+    end
+
+    def call
+      fetch_existing_files
+
+      begin
+        empty_files! if @empty_files.present?
+        http_upload! if @files.present?
+        remote_files! if @file_urls.present?
+
+        apply_commit_filelist
+      rescue APIError, StandardError => e
+        @errors << e.message
+      rescue Backend::Error => e
+        @errors << Xmlhash::XMLHash.new(error: e.summary)[:error]
+      end
+
+      @errors << 'No file or URI given' if @added_files.empty?
```

Comment:
You can move this to the beginning of the file in a [guard clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) fashion.

Reply:
That's true!